### PR TITLE
Create scrims from Discord

### PR DESCRIPTION
**Pull request by codeluggage**
- Adds a new `/scrim` command
- Improves slash command handling in the repo a little. Does the job for now until we refactor to use classes.

## Remaining steps

- [x] Resolve issue with scrim.new (or use scrimba.com/new)
- [x] Clean up layout of embed message

## Open questions

- What do we want the hint, title and description to say?
- Is `/scrim` the best naming?
- Do we want to show the less supported scrim templates, like Python, Typescript, etc?

### CODE REVIEW

File: src/ScrimIntegration.imba [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		client.on('interactionCreate', onInteractionCreate.bind(self))
+
+	def onReady client\Client
+		const { commands } = client.guilds.cache.first!
+		commands.create
+			name: 'scrim'
+			description: 'Create a new scrim to quickly teach or explain something'
+
+	def onInteractionCreate interaction
+		return unless interaction.isCommand! # NOTE: Hard to grok? Exclamation on a conditional
```
Comment (haikyuu) [inline src/ScrimIntegration.imba:31]: indeed

Comment (haikyuu) [inline src/ScrimIntegration.imba:31]: indeed

File: src/ScrimIntegration.imba [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		const command\BaseCommandInteraction = interaction # TODO: Use types better?
+
+		const reply = switch command.commandName
+			when 'scrim'
+				const embed = new MessageEmbed
+				embed.setTitle "Create a new scrim"
+				embed.setDescription "These are the scrim templates you can use - click one to try!"
+				# TODO: Read templates from DB? Cache to avoid hitting DB each time command is called?
```
Comment (haikyuu) [inline src/ScrimIntegration.imba:39]: Hardcoded is good. We rarely add or change them

File: src/ScrimIntegration.imba [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+				# TODO: Differentiate between servers and snowflake id's like emojis
+
+				const components = [
```
Comment (haikyuu) [inline src/ScrimIntegration.imba:42]: I'd change the ordering to:
- Standard
- React
- Vue
- Python
- Other (would go to the full list in scrimba.com)

**Review by haikyuu — COMMENTED** [review]
Sounds good. Let's just discuss the order and which templates to include

**Comment by perborgen** [comment]
I think `scrim` is the right word yes. 

Regarding naming:
Standard -> HTML, CSS, JavaScript

Wording for the title + description looks good to mw.

**Pull request merged**